tsl/test/expected/dist_distinct_pushdown.out

```diff
@@ -0,0 +1,132 @@
+-- This file and its contents are licensed under the Timescale License.
+-- Please see the included NOTICE for copyright information and
+-- LICENSE-TIMESCALE for a copy of the license.
+-- Need to be super user to create extension and add data nodes
+\c :TEST_DBNAME :ROLE_CLUSTER_SUPERUSER;
+\unset ECHO
+\set DATA_NODE_1 :TEST_DBNAME _1
+\set DATA_NODE_2 :TEST_DBNAME _2
+\set DATA_NODE_3 :TEST_DBNAME _3
+\set TABLESPACE_1 :TEST_DBNAME _1
+\set TABLESPACE_2 :TEST_DBNAME _2
+SELECT
+    test.make_tablespace_path(:'TEST_TABLESPACE1_PREFIX', :'TEST_DBNAME') AS spc1path,
+    test.make_tablespace_path(:'TEST_TABLESPACE2_PREFIX', :'TEST_DBNAME') AS spc2path
+\gset
+SELECT (add_data_node (name, host => 'localhost', DATABASE => name)).*
+FROM (VALUES (:'DATA_NODE_1'), (:'DATA_NODE_2'), (:'DATA_NODE_3')) v (name);
+          node_name          |   host    | port  |          database           | node_created | database_created | extension_created 
+-----------------------------+-----------+-------+-----------------------------+--------------+------------------+-------------------
+ db_dist_distinct_pushdown_1 | localhost | 55432 | db_dist_distinct_pushdown_1 | t            | t                | t
+ db_dist_distinct_pushdown_2 | localhost | 55432 | db_dist_distinct_pushdown_2 | t            | t                | t
+ db_dist_distinct_pushdown_3 | localhost | 55432 | db_dist_distinct_pushdown_3 | t            | t                | t
+(3 rows)
+
+GRANT USAGE ON FOREIGN SERVER :DATA_NODE_1, :DATA_NODE_2, :DATA_NODE_3 TO PUBLIC;
+-- Import testsupport.sql file to data nodes
+\unset ECHO
+-- Test DISTINCT ON pushdown.
+create table test_hypertable(ts timestamp, id int, val numeric);
+select create_hypertable('test_hypertable', 'ts');
+NOTICE:  adding not-null constraint to column "ts"
+      create_hypertable       
+------------------------------
+ (1,public,test_hypertable,t)
+(1 row)
+
+insert into test_hypertable select '2021-01-01 01:01:01'::timestamp + x * interval '1 second' ts,
+	mod(x, 4) id, r val
+from (select random() r, x from generate_series(1, 10000) x) tt
+order by r;
+create table test_distributed(ts timestamp, id int, val numeric);
+select create_distributed_hypertable('test_distributed', 'ts');
+NOTICE:  adding not-null constraint to column "ts"
+ create_distributed_hypertable 
+-------------------------------
+ (2,public,test_distributed,t)
+(1 row)
+
+insert into test_distributed select * from test_hypertable;
+-- The case with LIMIT serves as a reference.
+select ts, id from test_hypertable order by id, ts desc limit 1;
```

Comment:
It would be nice to see some explains here as well, especially to validate the remote SQL.

Reply:
Added.